contracts/src/libraries/PlonkVerifier.sol

```diff
@@ -171,84 +168,241 @@ library PlonkVerifier {
     }
 
     function _computeChallenges(
-        IPlonkVerifier.VerifyingKey memory verifyingKey,
-        uint256[] memory publicInput,
+        IPlonkVerifier.VerifyingKey memory vk,
+        uint256[] memory pi,
         IPlonkVerifier.PlonkProof memory proof
     ) internal pure returns (Challenges memory res) {
-        Transcript.TranscriptData memory transcript;
         uint256 p = BN254.R_MOD;
 
-        transcript.appendVkAndPubInput(verifyingKey, publicInput);
-
-        transcript.transcript = abi.encodePacked(
-            transcript.transcript,
-            proof.wire0.x,
-            proof.wire0.y,
-            proof.wire1.x,
-            proof.wire1.y,
-            proof.wire2.x,
-            proof.wire2.y
-        );
-        transcript.transcript = abi.encodePacked(
-            transcript.transcript, proof.wire3.x, proof.wire3.y, proof.wire4.x, proof.wire4.y
-        );
-
-        res.beta = transcript.getChallenge();
-        res.gamma = transcript.getChallenge();
-
-        transcript.transcript =
-            abi.encodePacked(transcript.transcript, proof.prodPerm.x, proof.prodPerm.y);
-
-        res.alpha = transcript.getChallenge();
-
-        transcript.transcript = abi.encodePacked(
-            transcript.transcript,
-            proof.split0.x,
-            proof.split0.y,
-            proof.split1.x,
-            proof.split1.y,
-            proof.split2.x,
-            proof.split2.y
-        );
-        transcript.transcript = abi.encodePacked(
-            transcript.transcript, proof.split3.x, proof.split3.y, proof.split4.x, proof.split4.y
-        );
+        assembly {
+            // use free memory space for scratch pad, 0x40: free memory ptr
+            let statePtr := mload(0x40)
+            let dataPtr := add(statePtr, 0x20)
+
+            // Start of transcript (unit: bytes)
+            // sizeInBits (4)  | domainSize (8) | numInputs (8) | pad (12)
```

Comment:
Should it be `sizeInBytes` instead of `sizeInBits`?

Reply:
sizeInBits which is why the value is 254. 
it's a data type of uint32 thus 4 bytes.

Reply:
I see, thanks.